cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/logging/Log.java

```diff
@@ -0,0 +1,125 @@
+package cloud.foundry.cli.crosscutting.logging;
+
+import java.util.logging.Level;
+
+/**
+ * The central logging module, providing a standardized interface for all logging activities in the project.
+ * By using this module instead of using custom logging facilities (like System.{out,err}.print*, for instance), a
+ * consistent style is ensured and log messages can be filtered properly.
+ * The interface is similar to Python's logging module, which provides a simple-to-use yet powerful API.
+ */
+public class Log {
+    // we can internally use the Java logging facilities
+    private static final java.util.logging.Logger logger;
+
+    // initialize and configure logger initially
+    static {
+        logger = java.util.logging.Logger.getLogger("cfctl");
+
+        // by default, we don't want to log debug messages
+        // however, the user can opt-in to them by setting the environment variable $DEBUG
+        if (System.getenv("DEBUG") != null) {
+            logger.setLevel(Level.FINE);
+        } else {
+            logger.setLevel(Level.INFO);
+        }
+    }
+
+    /**
+     * Internal helper building one string from multiple objects using a StringBuilder.
+     *
+     * @param arg0 mandatory log argument
+     * @param args optional additional arguments
+     */
+    private static String buildString(Object arg0, Object... args) {
+        StringBuilder message = new StringBuilder();
+
+        /*
+         requiring an initial arg to be passed and then allowing an array of args has two advantages:
+         - first of all, we ensure that we receive at least one argument
+         - second, we can easily ensure there's no unnecessary trailing whitespace, as we can append it before every
+           additional argument, as demonstrated in the loop below
+         */
+
+        message.append(arg0);
+
+        for (Object arg : args) {
+            message.append(" ");
+            message.append(arg);
+        }
+
+        return message.toString();
+    }
+
+    /**
+     * Log a message by a given loglevel. The message can consist of one or more objects.
+     *
+     * @param level loglevel to use
+     * @param arg0 mandatory log argument
+     * @param args optional additional arguments
+     */
+    private static void log(Level level, Object arg0, Object... args) {
+        String message = buildString(arg0, args);
+        logger.log(level, message);
+    }
+
+    /**
+     * Configure level of log messages to be shown to the user, e.g., by printing them to the console.
+     * @param level messages of this or any higher level will be shown by the logger
+     */
```

Comment:
Do we need this setter? 
It is set every time we call the .log methods anyways or?
Just curious it also doesnt harm anyone if not needed :)

Reply:
It'll be handy for testing. Also, I am not sure if the environment variable stuff is the best approach. I think it's good in principle, but actually the evaluation of `$DEBUG` should not be done in this class (SRP).